app/controllers/application_controller.rb

```diff
@@ -1490,8 +1489,20 @@ def apply_one_content_filter(fltr, query, user_filter)
     @any_content_filters_applied = true
   end
 
-  def show_index_count_results(query, args)
-    query.need_letters = args[:letters] if args[:letters]
+  ###########################################################################
+  #
+  # INDEX VIEW METHODS - MOVE TO HELPERS?
+
+  # These useful methods set a bunch of ivars used in all indexes.
+  # Converting it to a helper seems desirable re: separation of concerns,
+  # but we'd need access to the args passed to show_index_of_objects
+  # by the controller index action in the hypothetical helper.
+  # Also, these ivars are magically available to any index by default.
+  # I believe we'd have to add a helper manually to each index. - AN 2023
+  #
+  def set_index_view_ivars(query, args)
+    @error ||= :runtime_no_matches.t(type: query.model.type_tag)
+    @layout = calc_layout_params if args[:matrix]
     @timer_start = Time.current
```

Comment:
We could call it in the view with something like `set_view_content_for_index(@query, @index_args)`.

It would add an extra line at the top of every index view, but it would get that stuff out of the controller. 
Now that i think about it `@index_args` seems more like a legitimate use of an ivar, a concern proper to the controller.

Reply:
I'm not sure I have any good advice here.  I have the vague feeling that it all just a "big mess", but don't want to encourage big refactors.  Maybe when we go Sveltekit we can rethink the separation of concerns better at this level.  Or maybe piecemeal pulling things out of Application and starting to build something more in line with Trailblazer or some other better-considered framework is the way to go.  Really not sure how to approach it, because my memory is fuzzy on details, and it's overwhelming in scope.  I realize you're just asking about fiddling with the deck chairs surrounding show_index_of_objects, and your fiddle may well be an improvement.  If it makes sense to you and feels better to you, then I say go ahead.  One more thing: I think you can also rip out the number of results and query time.  It's just sort of embarrassing at this point.  Not sure anyone actually pays attention.

Reply:
Ok, thanks, will do that. 

Reading through the calls to `show_index_of_objects` i'm realizing that the `sorting_links` don't even have to pass through the `args` and get sent to `show_index_of_objects`. They can be directly defined in helpers per controller, like the tab set. 

Really hope when you look through the views/helpers in the future this type of organization will start to feel more straightforward, with to the refactors. It seems to help me think straight, anyway.